### ISSUE TITLE
Possible recursive expansion regression (3)

### ISSUE THREAD

**Issue by kioan000** [motdotla/dotenv-expand#123]
Sorry to bother you again 

While upgrading from 10.0 to latest 11.0.6

the following .env

```
FIRST_PAGE_URL=${PROJECT_PUBLIC_HOST}/first-page

...
MOCK_SERVER_HOST=http://localhost:${MOCK_SERVER_PORT}
MOCK_SERVER_PORT=8090

PROJECT_PUBLIC_HOST=${MOCK_SERVER_HOST}
```

parsed by this script

```
expand(dotenv.config({ path: path.resolve(root, '.env') }))

console.log('env', process.env)
```
is not recursively expanded anymore. Now it produces the following result:

```
FIRST_PAGE_URL: '${MOCK_SERVER_HOST}/first-page',
...
MOCK_SERVER_HOST: 'http://localhost:8090',
MOCK_SERVER_PORT: '8090',

PROJECT_PUBLIC_HOST='http://localhost:8090'


```

**Comment by motdotla** [comment]
this is working in newly released `v12.0.0` however with a warning:

https://github.com/motdotla/dotenv-expand/blob/master/tests/main.js#L602

**Comment by motdotla** [comment]
please note this behavior is different than in dotenvx, bash, docker-compose, etc. recommend switching to dotenvx if possible: https://github.com/dotenvx/dotenvx/pull/448
